### DISCUSSION TITLE
Navigating to function caller

### DISCUSSION THREAD

**Discussion by abcd-ca** [zed-industries/zed#6484] [Feature Requests]
Looking at a function call, I can quickly navigate to its definition by clicking cmd-clicking on it. Equally useful in my other IDE is being able to navigate backwards to find which places are calling the function. Note the "3 usages" link above my function signature here. When I click on that I get a list of files/locations I can click on to go up a parent level:

![image](https://user-images.githubusercontent.com/1296103/225050447-d863747c-4381-446f-a1e2-4dacc1d2fb24.png)
![image](https://user-images.githubusercontent.com/1296103/225051424-5da01777-94aa-4f9a-83ed-0a229e7f5351.png)

I don't see such a feature in Zed but it would be valuable. Note, it's much more ergonomic to navigate this way vs a shortcut that brings up a list of all usages in another panel that you have to later close.

**Comment by JosephTLyons** [comment]
Hey @abcd-ca, I'm assuming you already discovered the `Find All References` command, which does that.

Here is the issue we are using to track interest in being able to do this via a `cmd-click` on a definition:

- https://github.com/zed-industries/zed/issues/5351

Upvotes (👍) are appreciated.

**Comment by abcd-ca** [comment]
@JosephTLyons Thanks, I hadn't discovered it but I just tried it now. I tried it on a few TS types (functions, interfaces) and it didn't find all the uses – it would only find one. After I tried Find in Project for one of the interfaces' names and got many results, the Find All References shortcut did find all the references. I suspect it's something to do with lazy indexing of symbols.

**Comment by JosephTLyons** [comment]
`cmd-clicking` a symbol definition now launches a `find all references` action:

- https://github.com/zed-industries/zed/issues/5351
